README.md

```diff
@@ -614,3 +614,92 @@ with the corresponding inputs:
   params:
     paths: ['hello.html']
 ```
+
+### Action: `for_each`
+
+The `for_each` action lets you execute a sequence of steps repeatedly for each
+element of a list. For example, you might want your template to create several
+copies of a given file, one per application environment (e.g. production,
+staging).
+
+There are two variants of `for_each`. One variant accepts a hardcoded YAML list
+of values to iterate over in the `values` field. The other variant accepts a CEL
+expression in the `values_from` field that outputs a list of strings.
+
+Variant 1 example: hardcoded list of YAML values:
+
+```yaml
+- desc: 'Iterate over each (hard-coded) environment'
+  action: 'for_each'
+  params:
+    iterator:
+      key: 'env'
+      values: ['production', 'dev']
+    steps:
+      - desc: 'Do some action for each environment'
+        action: 'print'
+        params:
+          message: 'Now processing environment named {{.env}}'
+```
+
+Variant 2 example: a CEL expression that produces the list to iterate over:
+
+```yaml
+- desc: 'Iterate over each environment, produced by CEL as a list'
+  action: 'for_each'
+  params:
+    iterator:
+      key: 'env'
```

Comment:
I feel like `env` may not be the best name for an example variable, as its often used as a variable in similar workflow languages (like gh actions).